src/udev/udev-builtin-path_id.c

```diff
@@ -635,8 +636,24 @@ static int find_real_nvme_parent(sd_device *dev, sd_device **ret) {
                 return r;
 
         /* If the 'real parent' is (still) virtual, e.g. for nvmf disks, refuse to set ID_PATH. */
-        if (path_startswith(devpath, "/devices/virtual/"))
-                return -ENXIO;
+        if (path_startswith(devpath, "/devices/virtual/")) {
+		const char *syspath;
+		const char *addr, *fc_pn, *name, *nvme_pn;
+		struct sd_device *fcdev = NULL, *fc_parent = NULL;
+
+		sd_device_get_sysattr_value(nvme, "address", &addr);
+		nvme_pn = strrchr(addr, '-') + 1;
+
+		sd_device_new_from_syspath(&fc_parent, "/sys/class/fc_host");
+		FOREACH_DEVICE_CHILD_WITH_SUFFIX(fc_parent, child, name) {
+			fcdev = child;
+			sd_device_get_sysattr_value(fcdev, "port_name", &fc_pn);
+			if (strcmp(fc_pn, nvme_pn) == 0)
+				break;
+		}
+		sd_device_get_syspath(fcdev, &syspath);
```

Comment:
The native multipath nvme devices are based on FC adapters. FC device are only showed up one path in /dev/disk/by-path as well.  I think that the  devices with the second paths will be showed up in by-path if the first path failed because of udev daemon. I will double check. FYI: my environment is: nvmf over FC adapter. I created nvmf devices and fc devices in the same storage like v7k.

Reply:
""if (strcmp(fc_pn, nvme_pn) == 0)"
traddr=nn-0x50050768100035e5:pn-0x50050768101935e5,host_traddr=nn-0x200000109bec9b93:pn-0x100000109bec9b93
nvmf use fc's node name and port name to make connections.

Reply:
What I meant is that `syspath` is a loop-local variable which never gets used.

Reply:
I will re-work with error path.